org.eclipse.scanning.points/src/org/eclipse/scanning/points/CompoundIterator.java

```diff
@@ -56,6 +56,10 @@ private IPosition createFirstPosition() throws GeneratorException {
 	private IPosition next;
     private boolean justDidNext = false; // This attempts to deal with the case where someone does .next() without .hasNext();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)